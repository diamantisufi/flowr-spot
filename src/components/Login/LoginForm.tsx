import { TextInput } from "components/TextInput";
import React from "react";

type LoginFormProps = {
  onSubmit: (event: React.FormEvent<HTMLFormElement>) => void;
};

export const LoginForm: React.FC<LoginFormProps> = ({ onSubmit }) => {
  const status = "loading";

  return (
    <div className="flex flex-col">
      <form onSubmit={onSubmit}>
        <div className="mx-auto flex flex-col items-center justify-center gap-3">
          <TextInput
            label="Email Address"
            type="email"
            name="email"
            id="email"
            required
          />
          <TextInput
            type="password"
            name="password"
            id="password"
            minLength={6}
            required
            label="Password"
          />
          <button
            type="submit"
            className="button-primary w-full p-5 mt-3 text-white font-ubuntu font-medium"
          >
            {status === "loading" ? "Logging in..." : "Login to your Account"}
          </button>
        </div>
      </form>
    </div>
  );
};
