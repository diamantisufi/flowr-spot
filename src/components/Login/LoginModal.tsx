import React from 'react'
import Modal from 'react-modal'
import { LoginForm } from './LoginForm'
import { modalStyles } from 'common/modalStyles'

type LoginModalProps = {
  isOpen: boolean
  onClose: () => void
}

export const LoginModal: React.FC<LoginModalProps> = ({ isOpen, onClose }) => {
  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault()
    const formData = new FormData(event.currentTarget)

    const payload = {
      email: formData.get('email'),
      password: formData.get('password'),
    }
  }

  return (
    <Modal
      isOpen={isOpen}
      contentLabel="Login Modal"
      style={modalStyles}
      shouldCloseOnOverlayClick
      shouldCloseOnEsc
      onRequestClose={onClose}
      ariaHideApp={false}
    >
      <div className="modal-container">
        <h2 className="text-gray font-medium font-ubuntu mb-7">Welcome Back</h2>
      </div>
      <LoginForm onSubmit={handleSubmit} />
    </Modal>
  )
}
