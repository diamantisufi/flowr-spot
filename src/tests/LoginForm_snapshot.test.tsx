import { LoginForm } from 'components/Login/LoginForm'
import React from 'react'
import renderer from 'react-test-renderer'

describe('RegisterForm Snapshot', () => {
  it('matches snapshot', () => {
    const mockOnSubmit = jest.fn()
    const mockError = ''

    const tree = renderer
      .create(
        <LoginForm onSubmit={mockOnSubmit} error={mockError} loading={false} />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
