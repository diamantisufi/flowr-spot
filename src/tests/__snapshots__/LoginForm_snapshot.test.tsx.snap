// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterForm Snapshot matches snapshot 1`] = `
<div
  className="flex flex-col"
>
  <form
    data-testid="login-form"
    onSubmit={[MockFunction]}
  >
    <div
      className="mx-auto flex flex-col items-center justify-center gap-3"
    >
      <div
        className="relative rounded flex flex-col w-full h-full border border-inputBorder bg-inputBackground hover:border-slate-300 transition-colors text-[13px]"
      >
        <label
          className="absolute top-2 text-[10px] text-secondary ml-4"
          htmlFor="email"
        >
          Email Address
        </label>
        <input
          className="outline-none h-full w-full bg-transparent text-gray p-4 pt-7 undefined"
          id="email"
          name="email"
          required={true}
          type="email"
        />
      </div>
      <div
        className="relative rounded flex flex-col w-full h-full border border-inputBorder bg-inputBackground hover:border-slate-300 transition-colors text-[13px]"
      >
        <label
          className="absolute top-2 text-[10px] text-secondary ml-4"
          htmlFor="password"
        >
          Password
        </label>
        <input
          className="outline-none h-full w-full bg-transparent text-gray p-4 pt-7 undefined"
          id="password"
          minLength={6}
          name="password"
          required={true}
          type="password"
        />
      </div>
      <button
        className="button-primary w-full p-5 mt-3 text-white font-ubuntu font-medium"
        disabled={false}
        type="submit"
      >
        Login to your Account
      </button>
    </div>
  </form>
</div>
`;
